Struct fields supported for header and path param types

tanenbaum opened the pull request:
**Describe the PR**
Supports "object" (i.e. struct model) data types for `header` and `path` parameter types via `@Param` operation comment in the same way that `query` and `formData` already work - basic structs are supported to represent multiple paramters at once.

This allows struct models, e.g. for a number of related headers, to be used directly rather than duplicating all the `@Param` comments for every field in those structs.

This works similarly to existing `formData` and `query` types and largely reuses all the same operation parsing logic.

In addition this supports using struct tags `header:"name"`  and `uri:"name"` for `header` and `path` params respectively. This follows the Gin specific tag naming conventions.

**Relation issues**
A few issues with similar feature requests:
- https://github.com/swaggo/swag/issues/248
- https://github.com/swaggo/swag/issues/1720
- https://github.com/swaggo/swag/issues/1197 (closed in error I believe)

tanenbaum commented:
Looking at rebasing this and fixing the conflicts - I'm not sure if the change `return strings.TrimRight(strings.TrimSpace(strings.Split(ps.tag.Get(formTag), ",")[0]), "[]")` is in fact backwards compatible - what about API definitions where people want to include the `[]` characters in their parameter names?

sdghchj commented:
[] is trivial, make nonsense.&nbsp;
trim them.



------------------ Original ------------------
From: Joe Shaw ***@***.***&gt;
Date: Thu,Jan 18,2024 7:48 PM
To: swaggo/swag ***@***.***&gt;
Cc: sdghchj ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [swaggo/swag] Struct fields supported for header and path paramtypes (PR #1740)





 
Looking at rebasing this and fixing the conflicts - I'm not sure if the change introduced in #1706 return strings.TrimRight(strings.TrimSpace(strings.Split(ps.tag.Get(formTag), ",")[0]), "[]") is in fact backwards compatible - what about API definitions where people want to include the [] characters at the end of their parameter names? While not common, it could cause unwanted doc changes.
 
@sdghchj what do you think about this? Should all the fields trim "[]"? I'm not so sure.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

tanenbaum commented:
> [] is trivial, make nonsense.&nbsp; trim them.
> […](#)
> ------------------ Original ------------------ From: Joe Shaw ***@***.***&gt; Date: Thu,Jan 18,2024 7:48 PM To: swaggo/swag ***@***.***&gt; Cc: sdghchj ***@***.***&gt;, Mention ***@***.***&gt; Subject: Re: [swaggo/swag] Struct fields supported for header and path paramtypes (PR #1740) Looking at rebasing this and fixing the conflicts - I'm not sure if the change introduced in #1706 return strings.TrimRight(strings.TrimSpace(strings.Split(ps.tag.Get(formTag), ",")[0]), "[]") is in fact backwards compatible - what about API definitions where people want to include the [] characters at the end of their parameter names? While not common, it could cause unwanted doc changes. @sdghchj what do you think about this? Should all the fields trim "[]"? I'm not so sure. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

Fair enough - rebased and updated.
